pkg/dockerregistry/server/metrics/oapi.go

```diff
@@ -0,0 +1,199 @@
+}
+
+func (c *oapiClientImageStreamImages) Get(name, id string) (*imageapi.ImageStreamImage, error) {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreamimages.get"}).Stop()
+	return c.client.Get(name, id)
+}
+
+type oapiClientImageStreams struct {
+	client oapi.ClientImageStreamInterface
+}
+
+func (c *oapiClientImageStreams) Get(name string, options metav1.GetOptions) (*imageapi.ImageStream, error) {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreams.get"}).Stop()
+	return c.client.Get(name, options)
+}
+
+func (c *oapiClientImageStreams) Create(stream *imageapi.ImageStream) (*imageapi.ImageStream, error) {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreams.create"}).Stop()
+	return c.client.Create(stream)
+}
+
+type oapiClientImageStreamMappings struct {
+	client oapi.ClientImageStreamMappingInterface
+}
+
+func (c *oapiClientImageStreamMappings) Create(mapping *imageapi.ImageStreamMapping) error {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreammapping.create"}).Stop()
+	return c.client.Create(mapping)
+}
+
+type oapiClientImageStreamSecrets struct {
+	client oapi.ClientImageStreamSecretInterface
+}
+
+func (c *oapiClientImageStreamSecrets) Secrets(name string, options metav1.ListOptions) (*kapi.SecretList, error) {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreamsecrets.secrets"}).Stop()
+	return c.client.Secrets(name, options)
+}
+
+type oapiClientImageStreamTags struct {
+	client oapi.ClientImageStreamTagInterface
+}
+
+func (c *oapiClientImageStreamTags) Delete(name, tag string) error {
+	defer NewTimer(MasterAPIRequests, []string{"imagestreamtags.delete"}).Stop()
+	return c.client.Delete(name, tag)
+}
+
+type oapiClientLimitRanges struct {
+	client oapi.ClientLimitRangeInterface
+}
+
+func (c *oapiClientLimitRanges) List(opts metav1.ListOptions) (*kapi.LimitRangeList, error) {
+	defer NewTimer(MasterAPIRequests, []string{"limitranges.list"}).Stop()
+	return c.client.List(opts)
+}
+
+type oapiClientUsers struct {
+	client oapi.ClientUserInterface
+}
+
+func (c *oapiClientUsers) Get(name string, options metav1.GetOptions) (*userapi.User, error) {
+	defer NewTimer(MasterAPIRequests, []string{"users.get"}).Stop()
+	return c.client.Get(name, options)
+}
+
+type oapiClientLocalSubjectAccessReviews struct {
+	client oapi.ClientLocalSubjectAccessReviewInterface
+}
+
+func (c *oapiClientLocalSubjectAccessReviews) Create(policy *authorizationapi.LocalSubjectAccessReview) (*authorizationapi.SubjectAccessReviewResponse, error) {
+	defer NewTimer(MasterAPIRequests, []string{"localsubjectaccessreviews.create"}).Stop()
+	return c.client.Create(policy)
+}
+
+type oapiClientSubjectAccessReviews struct {
+	client oapi.ClientSubjectAccessReviewInterface
+}
+
+func (c *oapiClientSubjectAccessReviews) Create(policy *authorizationapi.SubjectAccessReview) (*authorizationapi.SubjectAccessReviewResponse, error) {
+	defer NewTimer(MasterAPIRequests, []string{"subjectaccessreviews.create"}).Stop()
```

Comment:
@smarterclayton We need to see how many times (and how long it takes) the registry server requests the master API for each user request. Can we do it with metrics from he kubernetes client ?